android/src/main/java/com/polidea/reactnativeble/advertisement/AdvertisementData.java

```diff
@@ -0,0 +1,155 @@
+package com.polidea.reactnativeble.advertisement;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class AdvertisementData {
+
+    private byte[] manufacturerData;
+    private Map<UUID, byte[]> serviceData;
+    private ArrayList<UUID> serviceUUIDs;
+    private Integer txPowerLevel;
+    private ArrayList<UUID> solicitedServiceUUIDs;
+
+    static long BLUETOOTH_BASE_UUID_LSB = 0x800000805F9B34FBL;
+    static int  BLUETOOTH_BASE_UUID_MSB = 0x00001000;
+
+    public byte[] getManufacturerData() {
+        return manufacturerData;
+    }
+
+    public Map<UUID, byte[]> getServiceData() {
+        return serviceData;
+    }
+
+    public ArrayList<UUID> getServiceUUIDs() {
+        return serviceUUIDs;
+    }
+
+    public Integer getTxPowerLevel() {
+        return txPowerLevel;
+    }
+
+    public ArrayList<UUID> getSolicitedServiceUUIDs() {
+        return solicitedServiceUUIDs;
+    }
+
+    public static AdvertisementData parseScanResponseData(byte[] advertisement) {
+        AdvertisementData advData = new AdvertisementData();
+        ByteBuffer rawData = ByteBuffer.wrap(advertisement).order(ByteOrder.LITTLE_ENDIAN);
+        while (rawData.remaining() >= 2) {
+            int adLength = (rawData.get() & 0xFF) - 1;
```

Comment:
I'm not really sure about it. Some field allow to have 0 octets of data. For example flags:

> The Flags field may be zero or more octets long. This allows the Flags field to be extended while using the minimum number of octets within the data packet.

Also with this code it would be not possible to have empty arrays of UUIDs and there
is an example on page 26 in Supplement to Core Bluetooth Specification which shows
exactly this behaviour.

Corresponding functions will check theirs size per type.